CDTDatastoreTests/CDTReplicationTests.m

```diff
@@ -569,13 +497,17 @@ -(void)testForProhibitedOptionalReplicationHeaders
 {
     CDTPullReplication *pull;
     NSError *error;
-    NSDictionary *pullDoc;
+    NSMutableDictionary *pullDoc;
```

Comment:
As above, remove `pullDoc` and see if there is anything to be gained from asserts which compare the various settings (headers, interceptors) between TD and CDT replicators.